x-pack/plugins/apm/public/components/app/service_overview/service_overview_charts/latency_map/embedded_map.tsx

```diff
@@ -0,0 +1,177 @@
+import { MAP_SAVED_OBJECT_TYPE } from '@kbn/maps-plugin/common';
+import {
+  ErrorEmbeddable,
+  ViewMode,
+  isErrorEmbeddable,
+  EmbeddableFactory,
+} from '@kbn/embeddable-plugin/public';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import { EuiText } from '@elastic/eui';
+import type { Filter } from '@kbn/es-query';
+import { ApmPluginStartDeps } from '../../../../../plugin';
+import { getLayerList } from './get_layer_list';
+import { useApmParams } from '../../../../../hooks/use_apm_params';
+import { useTimeRange } from '../../../../../hooks/use_time_range';
+
+export function EmbeddedMapComponent({ filters }: { filters: Filter[] }) {
+  const {
+    query: { rangeFrom, rangeTo, kuery },
+  } = useApmParams('/services/{serviceName}/overview');
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+  const [error, setError] = useState<boolean>();
+
+  const [embeddable, setEmbeddable] = useState<
+    MapEmbeddable | ErrorEmbeddable | undefined
+  >();
+
+  const embeddableRoot: React.RefObject<HTMLDivElement> =
+    useRef<HTMLDivElement>(null);
+
+  const {
+    embeddable: embeddablePlugin,
+    maps,
+    notifications,
+  } = useKibana<ApmPluginStartDeps>().services;
+
+  useEffect(() => {
+    async function setupEmbeddable() {
+      const factory:
+        | EmbeddableFactory<
+            MapEmbeddableInput,
+            MapEmbeddableOutput,
+            MapEmbeddable
+          >
+        | undefined = embeddablePlugin?.getEmbeddableFactory(
+        MAP_SAVED_OBJECT_TYPE
+      );
+
+      if (!factory) {
+        setError(true);
+        notifications?.toasts.addDanger({
+          title: i18n.translate(
+            'xpack.apm.serviceOverview.embeddedMap.error.toastTitle',
+            {
+              defaultMessage: 'An error occurred when adding map embeddable',
+            }
+          ),
+          text: i18n.translate(
+            'xpack.apm.serviceOverview.embeddedMap.error.toastDescription',
+            {
+              defaultMessage: `Embeddable factory with id "{embeddableFactoryId}" was not found.`,
+              values: {
+                embeddableFactoryId: MAP_SAVED_OBJECT_TYPE,
+              },
+            }
+          ),
+        });
+      }
+
+      const input: MapEmbeddableInput = {
+        attributes: { title: '' },
+        id: uuid.v4(),
+        title: i18n.translate(
+          'xpack.apm.serviceOverview.embeddedMap.input.title',
+          {
+            defaultMessage: 'Latency by country',
+          }
+        ),
+        filters,
+        viewMode: ViewMode.VIEW,
+        isLayerTOCOpen: false,
+        query: {
+          query: kuery,
+          language: 'kuery',
+        },
+        timeRange: {
+          from: start,
+          to: end,
+        },
+        hideFilterActions: true,
+      };
+
+      const embeddableObject = await factory?.create(input);
```

Comment:
Like the comment above, you wouldn't need to add `?` after factory here if you had stoped the function or only run this peace of code in the `else` block when factory is defined.